src/main.cpp

```diff
@@ -4065,7 +4066,8 @@ bool CheckBlock(const CBlock& block, CValidationState& state, bool fCheckPOW, bo
 
     if (block.IsProofOfStake()) {
         // Coinbase output should be empty if proof-of-stake block
-        if (block.vtx[0].vout.size() != 1 || !block.vtx[0].vout[0].IsEmpty())
+        int commitpos = GetWitnessCommitmentIndex(block);
+        if (block.vtx[0].vout.size() != (commitpos == -1 ? 1 : 2) || !block.vtx[0].vout[0].IsEmpty())
```

Comment:
Shouldn't `!block.vtx[0].vout[0].IsEmpty()` be changed to allow a segwit commitment?

Reply:
I think for segwit block.vtx[0].vout[0] should be empty and block.vtx[0].vout[1].IsEmpty() should not be empty?